unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+ * @test_id             controller_commit_changes_failed
+ * @brief               Test case of controller_commit_changes() where commit_changes() fails, return -1
+ *                      and wl_resource_get_user_data() success, return an object
+ * @test_procedure Steps:
+ *                      -# Mocking the commit_changes() does return -1
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_commit_changes()
+ *                      -# Verification point:
+ *                         +# commit_changes() must be called once time
+ */
+TEST_F(ControllerTests, controller_commit_changes_failed)
+{
+    SET_RETURN_SEQ(commit_changes, mp_failureResult, 1);
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+
+    controller_commit_changes(nullptr, nullptr);
+
+    ASSERT_EQ(commit_changes_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_commit_changes_success
+ * @brief               Test case of controller_commit_changes() where commit_changes() success, return 0
+ *                      and wl_resource_get_user_data() success, return an object
+ * @test_procedure Steps:
+ *                      -# Mocking the commit_changes() does return 0
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_commit_changes()
+ *                      -# Verification point:
+ *                         +# commit_changes() must be called once time
+ */
+TEST_F(ControllerTests, controller_commit_changes_success)
+{
+    SET_RETURN_SEQ(commit_changes, mp_successResult, 1);
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+
+    controller_commit_changes(nullptr, nullptr);
+
+    ASSERT_EQ(commit_changes_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_set_surface_visibility_wrongLayoutSurface
+ * @brief               Test case of controller_set_surface_visibility() where get_surface_from_id() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_set_surface_visibility()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_set_surface_visibility_wrongLayoutSurface)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+
+    controller_set_surface_visibility(nullptr, nullptr, 1, 1);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
```

Comment:
should verify wl_resource_post_event_fake.arg1_history and surface_set_visibility.call_count